Add standardized sources and targets definitions across dataset modules

amir9667 opened the pull request:
Fixes #1283 

This pull request introduces  definition and integration of class-level sources and targets for the affected dataset modules.
references to cfg have been removed and replaced with static, structured access through DatasetSources and DatasetTargets.

**Key Changes**

    1- Added fully structured sources and targets attributes to each updated dataset class.
    2-Replaced all cfg[...] usages with direct references to
    ClassName.sources.files[...],
    ClassName.sources.tables[...], or
    ClassName.targets[...].


🧑‍💻 Contributor Checklist

Before requesting a review, make sure you've completed all of the following:

- [ ] All tests pass locally or via CI
(for more information on local test, check tox in the [Contributing section](https://egon-data.readthedocs.io/en/latest/contributing.html#))
(CI tests are automatically executed when creating a PR, you can see the results of the checks below)
- [x] Workflow has run at least once in Test mode
(optional if no dataset changes are involved)
- [ ] Relevant documentation is updated (API, new features, etc.)
- [x] Dataset-versions are updated when existing datasets are adjusted.
- [x] Added a note to CHANGELOG.rst about the changes
- [ ]     Added yourself to AUTHORS.rst

Optional:

- [ ] Changes have been tested in Everything mode
- [ ] Extend the checklist for reviewers: Which aspects should be reviewed in particular?

🔍 Reviewer Checklist

During your review, please check the following:

- [x] Is the code clean, readable, and efficient? Are there any oddities or obvious inefficiencies?
- [x] Does the code work as expected? (should already be verified by contributor)
- [ ] Do all tests pass? (see CI results)
- [ ] Is the documentation complete and up to date?
- [x] Is CHANGELOG.rst updated accordingly?
- [ ] Is all necessary metadata complete and correct?
         - [ ] If metadata is pending: Is there an appropriate issue filed?

amir9667 committed:
Fixing the Errors

amir9667 committed:
Fixing the Errors

mheshammenisy committed:
docs(osm): update download() comment

mheshammenisy committed:
fix(zensus): adjust sources and targets definitions

mheshammenisy committed:
fix(heat_demand_timeseries): correct SQL query syntax

mheshammenisy committed:
fix(era5): adjust sources and targets attributes

mheshammenisy committed:
fix(etrago_setup): adjust sources and targets attributes

mheshammenisy committed:
fix(gas_areas): adjust sources and targets attributes

mheshammenisy committed:
fix(industrial_gas_demand): adjust sources and targets attributes

mheshammenisy committed:
fix(loadarea): adjust sources and targets attributes

mheshammenisy committed:
fix(osmtgmod): adjust sources and targets attributes

mheshammenisy committed:
fix(re_potential_areas): adjust sources and targets attributes

mheshammenisy committed:
fix(renewable_feedin): adjust sources and targets attributes

mheshammenisy committed:
fix(scenario_parameters): adjust sources and targets attributes

mheshammenisy committed:
fix(substation_voronoi): adjust sources and targets attributes

mheshammenisy committed:
fix(tyndp): adjust sources and targets attributes

mheshammenisy committed:
fix: clean up sources/targets attributes and references in Mastr

mheshammenisy committed:
fix: clean up sources/targets attributes and references in SanityChecks

mheshammenisy committed:
add sources and targets for bus.py

mheshammenisy committed:
add sources and targets for h2_grid.py

mheshammenisy committed:
add sources and targets for h2_to_ch4.py

mheshammenisy committed:
add sources and targets for power_to_h2.py

mheshammenisy committed:
add sources and targets for storage.py

mheshammenisy committed:
update sources and targets in __init__.py

mheshammenisy committed:
add sources and targets for power_to_heat.py

mheshammenisy committed:
add sources and targets for hts_etrago.py

mheshammenisy committed:
updating sources in __init__.py

mheshammenisy committed:
adjust sources and target attributes in gas_grid.py

mheshammenisy committed:
adjust sources and target attributes in fill_etrago_gen.py

mheshammenisy committed:
adjust sources and target attributes in electrical_neighbours.py

amir9667 committed:
Fixed errors

amir9667 committed:
updating souce and targets

amir9667 committed:
Fixing the Errors

mheshammenisy committed:
fix: correct sources file path in electrical_neighbours

mheshammenisy committed:
fix: sources and targets imports

mheshammenisy committed:
fix: use HeatEtrago DatasetSources/Targets in power_to_heat

mheshammenisy committed:
fix: use DatasetSources/Targets for hydrogen_etrago modules

mheshammenisy committed:
etrago_hydrogen config in hydrogen bus

nesnoj commented:
Looking forward to this nice feature! :eyes:

amir9667 committed:
Fixing the error and refactoring load sources

ClaraBuettner commented on src/egon/data/datasets/emobility/.spyproject/config/defaults/defaults-codestyle-0.2.0.ini:
```diff
@@ -0,0 +1,5 @@
+indentation = True
+edge_line = True
+edge_line_columns = 79
+
```
Please remove the .ini files, also in the other subdirectories (they are only used locally and shouldn't be part of the repository).

ClaraBuettner commented on src/egon/data/datasets/heat_etrago/hts_etrago.py:
```diff
@@ -6,14 +6,14 @@
-    sources = config.datasets()["etrago_heat"]["sources"]
-    targets = config.datasets()["etrago_heat"]["targets"]
+    sources = HtsEtragoTable.sources.tables
+    targets = HtsEtragoTable.targets.tables
```
Are these line working without importing HtsEtragoTable? 
Just by looking at the code, I think you either need to import it or use the `load_sources_and_targets` function from `egon.data.datasets` as you e.g. do it in heat_supply/district_heating.py

ClaraBuettner commented on src/egon/data/datasets/heat_etrago/power_to_heat.py:
```diff
@@ -23,17 +25,17 @@ def insert_individual_power_to_heat(scenario):
-    sources = config.datasets()["etrago_heat"]["sources"]
-    targets = config.datasets()["etrago_heat"]["targets"]
+    #sources = config.datasets()["etrago_heat"]["sources"]
+    #targets = config.datasets()["etrago_heat"]["targets"]
```
Commented code could be confusing for other developers, because they don't know why it was commented. 
So please remove code that is not needed anymore instead of commenting it.

ClaraBuettner commented on src/egon/data/datasets/osm/__init__.py:
```diff
@@ -71,26 +69,26 @@ def to_postgres(cache_size=4096):
     ) as p:
         shutil.copy(p, style_file)
-
+        
```
Before merging your branch, some code style tools should be applied.

ClaraBuettner commented on src/egon/data/datasets/heat_supply/__init__.py:
```diff
@@ -391,6 +389,28 @@ class HeatSupply(Dataset):
+            "map_vg250_grid": "boundaries.egon_map_mvgriddistrict_vg250",
+            "mv_grids": "grid.egon_mv_grid_district",
+            "map_dh": "demand.egon_map_zensus_district_heating_areas",
+            "etrago_buses": "grid.egon_etrago_bus",
```
Table sources are sometimes defined like this and sometimes with a sub-dictionary {"table": ..., "schema": ...}. Are both options working the same way?

mheshammenisy committed:
add sources and targets for ch4_prod.py

mheshammenisy committed:
add sources and targets for district_heating_areas.__init__.py

mheshammenisy committed:
add sources and targets for electricity_demand_etrago.py

mheshammenisy committed:
update sources in heat_demand_timeseries.__init__.py

mheshammenisy committed:
import sources and targets for daily.py

mheshammenisy committed:
import sources and targets for idp_pool.py

mheshammenisy committed:
import sources and targets for service_sector.py

mheshammenisy committed:
update sources and targets attributes for power_to_heat.py

mheshammenisy committed:
update importing sources and targets from __init__ in bus.py

mheshammenisy committed:
update sources and targets import in h2_grid.py

mheshammenisy committed:
update sources and targets import in power_to_h2.py

mheshammenisy committed:
update sources and targets import in storage.py

mheshammenisy committed:
import sources and targets in parameters.py

mheshammenisy committed:
update sources and targets attributes for ch4_prod.py

mheshammenisy committed:
fix path in electricity_demand_etrago.py

mheshammenisy committed:
fix sources in heat_demand_timeseries/__init__.py

mheshammenisy committed:
fix sources and targets attributes in idp_pool.py

mheshammenisy committed:
fix import error dataset

mheshammenisy committed:
fix imports in heat_demand_timeseries modules

mheshammenisy committed:
fix sources bug in electricity_demand_etrago.py

mheshammenisy committed:
fix syntax error in service_sector.py

mheshammenisy committed:
fix: load sources/targets inside functions (bus.py)

mheshammenisy committed:
fix: load sources/targets inside functions (h2_grid.py)

mheshammenisy committed:
fix: load sources/targets inside functions (h2_to_ch4.py)

mheshammenisy committed:
fix: load sources/targets inside functions (power_to_h2.py)

mheshammenisy committed:
fix: load sources/targets inside functions (storage.py)

mheshammenisy committed:
fix: load sources/targets inside functions (daily.py)

mheshammenisy committed:
fix: load sources/targets inside functions (idp_pool.py)

mheshammenisy committed:
fix: load sources/targets inside functions (service_sector.py)

mheshammenisy committed:
fix: load sources/targets inside functions (power_to_heat.py)

mheshammenisy committed:
fix: update sources (osmtgmod/__init__.py)

mheshammenisy committed:
fix: import load_sources_and_targets from datasets (substation.py)

amir9667 committed:
Revert "Fixing the error and refactoring load sources"

This reverts commit 2621beda0e667d8871ad79bb3991fba57225b706.

amir9667 committed:
fixing the error related to chp.insert

amir9667 committed:
Fixing the errors and refactoring the remainings

amir9667 committed:
Refactored

amir9667 committed:
Refactored and fixed the airflow's error

amir9667 committed:
errors fixed

amir9667 committed:
fixing of airflow's error

amir9667 committed:
resolving init files

amir9667 committed:
resolving airflow error

amir9667 committed:
resolving airflow error

amir9667 committed:
same: fixing errors!

mheshammenisy committed:
fix: restore dataset config for H2 grid input files

mheshammenisy committed:
fix: add sources and targets for mv_grid_districts

amir9667 committed:
Fixing airflow error

amir9667 committed:
Fixing the airflow error

amir9667 committed:
going back to initial version to see if the error is there or not

amir9667 committed:
restoring because the error is due to the initial code itself

amir9667 committed:
refactoring the remainings based on load_source_target import

amir9667 committed:
refactoring the remainings based on load_source_target import

amir9667 committed:
Fixing the airflow's error

amir9667 committed:
Fixing the airflow's error

amir9667 committed:
Fixing the airflow's error

amir9667 committed:
refactoring the remainings based on load_source_target import

amir9667 committed:
refactoring the remainings of CHP based on load_source_target import

amir9667 committed:
Fixing the airflow's error

amir9667 committed:
Fixing the airflow's error

amir9667 committed:
Fixing the airflow's error

amir9667 committed:
Back to previous version

mheshammenisy committed:
fix(heat_demand): remove legacy configuration comments from __init__

mheshammenisy committed:
fix(saltcavern): remove legacy configuration comments from __init__

mheshammenisy committed:
fix(scenario_parameters): remove legacy configuration comments from __init__

mheshammenisy committed:
fix(scenario_parameters): load sources and targets inside function

mheshammenisy committed:
fix(society_prognosis): remove legacy configuration comments

mheshammenisy committed:
vg250_mv_grid_districts: define DatasetSources and DatasetTargets

mheshammenisy committed:
zensus_mv_grid_districts: define DatasetSources and DatasetTargets

mheshammenisy committed:
zensus_vg250: define DatasetSources and DatasetTargets

mheshammenisy committed:
power_etrago/__init__: add DatasetSources and DatasetTargets

mheshammenisy committed:
match_ocgt: load DatasetSources and DatasetTargets

mheshammenisy committed:
electricity_demand/__init__: add DatasetSources and DatasetTargets

mheshammenisy committed:
electricity_demand/temporal: load DatasetSources and DatasetTargets

mheshammenisy committed:
cts_buildings: define DatasetSources and DatasetTargets

mheshammenisy committed:
hh_buildings: define DatasetSources and DatasetTargets

mheshammenisy committed:
hh_profiles: define DatasetSources and DatasetTargets

amir9667 committed:
refactoring the remainings of storage based on load_source_target import

amir9667 committed:
Fixing the airflow's error

amir9667 committed:
Fixing the airflow's error

amir9667 committed:
refactoring the remainings based on load_source_target import

mheshammenisy committed:
fix: electricity demand temporal sources/targets access

mheshammenisy committed:
fix: SQL f-string syntax in heat demand time series

amir9667 committed:
refactoring the remainings based on load_source_target import

mheshammenisy committed:
chore: bump version in GasAreas

mheshammenisy committed:
chore: bump version in HydrogenEtrago

mheshammenisy committed:
fix: correct schema.table SQL formatting in hydrogen bus

amir9667 committed:
go back to dev version to spot the error

amir9667 committed:
error fixing in airflow

amir9667 committed:
error fixing in airflow

amir9667 committed:
error fixing in airflow

mheshammenisy committed:
fix: clean up sources/targets references in GasAreas

mheshammenisy committed:
fix: update sources/targets definitions in hydrogen_etrago init

mheshammenisy committed:
fix: clean up sources/targets references in HydrogenBusEtrago

mheshammenisy committed:
fix: clean up sources/targets references in HydrogenStorageEtrago

mheshammenisy committed:
refactor: define sources and targets for scenario capacities

mheshammenisy committed:
fix: refactor storage module in hydrogen_etrago

amir9667 committed:
Refactoring the MIT charging infrustructure

amir9667 committed:
fixing airflow's error

amir9667 committed:
Remove local .spyproject config files

amir9667 committed:
Remove local .spyproject config files from chp

amir9667 committed:
Remove local .spyproject config files from gas_neighbours

amir9667 committed:
Remove local .spyproject config files from industrial_sites

amir9667 committed:
Remove local .spyproject config files from substation

mheshammenisy committed:
chore: temporarily restore config-based access in saltcavern write task

mheshammenisy committed:
fix: clean up VG250 sources/targets and bump dataset version

mheshammenisy committed:
chore: bump DemandRegio version

mheshammenisy committed:
temp: restore old config for h2_grid

mheshammenisy committed:
temp: restore old config for h2_to_ch4

mheshammenisy committed:
temp: restore old config for power_to_h2

mheshammenisy committed:
temp: restore old config for storage

mheshammenisy committed:
fix: next_etrago_id bug

mheshammenisy committed:
fix: next_etrago_id bug (second occurrence)

mheshammenisy committed:
temp: restore old config for match_ocgt

mheshammenisy committed:
fix(heat_etrago/__init__.py): adjust sources attribute

mheshammenisy committed:
fix(heat_etrago): fix sources attribute table reference (geom_polygon)

mheshammenisy committed:
bump: HeatSupply dataset version for Airflow rerun

amir9667 committed:
restoring the sub code to update the dataset

amir9667 committed:
Fixing the error

amir9667 committed:
Fixing the error

amir9667 committed:
Fixing the error

amir9667 committed:
switching to cfg version t get the dataset

amir9667 committed:
refactoring MIT infrastructure

amir9667 committed:
refactoring infrastructure allocation

mheshammenisy committed:
temp: restore old cfg usage in district_heating for execution order

amir9667 committed:
refactoring db classes

ClaraBuettner committed:
Update dataset sources and targets before all tasks are executed

mheshammenisy committed:
chore: bump version to rerun CTS buildings dataset

mheshammenisy committed:
temp: restore old cfg in individual heating to allow rerun

mheshammenisy committed:
chore: bump version in individual heating dataset

amir9667 committed:
Merge remote-tracking branch 'origin/features/#1283-update-source-and-targets-early' into features/#1283-fill-dataset-sources-and-targets

mheshammenisy committed:
chore: bump dataset versions

mheshammenisy committed:
fix: move self.register_sources_and_targets to Dataset initialization

mheshammenisy committed:
fix: indentation

amir9667 committed:
Fixing the error

amir9667 committed:
Fixing the error

amir9667 committed:
Fixing the error

amir9667 committed:
Fixing the error

amir9667 committed:
Fixing the error

amir9667 committed:
Fixing the error

amir9667 committed:
Fixing the error

amir9667 committed:
Fixing the error

amir9667 committed:
Fixing the error

amir9667 committed:
Refactoring Emobility_heavy_duty

amir9667 committed:
fixing the error

amir9667 committed:
fixing the error

amir9667 committed:
fixing the error

amir9667 committed:
fixing the error

amir9667 committed:
fixing the error

mheshammenisy committed:
fix: load sources and targets in sanity checks

mheshammenisy committed:
fix: restore sources and targets refactor district_heating

mheshammenisy committed:
fix: restore sources and targets refactor individual_heating

mheshammenisy committed:
fix: restore sources and targets refactor h2_grid

mheshammenisy committed:
fix: restore sources and targets refactor h2_to_ch4

mheshammenisy committed:
fix: restore sources and targets refactor power_to_h2

mheshammenisy committed:
fix: restore sources and targets refactor storage

mheshammenisy committed:
fix: restore sources and targets refactor match_ocgt

amir9667 committed:
fixing the airflow's error

amir9667 committed:
fixing the airflow's error

amir9667 committed:
fixing the airflow's error

mheshammenisy committed:
bump: version HouseholdElectricityDemand to 0.0.7

mheshammenisy committed:
fix: bump DemandRegio version

mheshammenisy committed:
fix: dataset bug registration sources and targets

mheshammenisy committed:
refactor: general update sources and targets

mheshammenisy committed:
fix: adjust register_sources_and_targets logic

mheshammenisy committed:
bump: district_heating_areas version

mheshammenisy committed:
refactor: update sources and targets definitions

mheshammenisy committed:
fix: dataset name typo

mheshammenisy committed:
update sources and targets definitions

mheshammenisy committed:
fix: refactor bug in HydrogenBusEtrago

mheshammenisy committed:
refactor: refine DatasetSources and DatasetTargets definitions and references

mheshammenisy committed:
chore: bump SubstationExtraction dataset version

mheshammenisy committed:
fix: bump versions

mheshammenisy committed:
bump:  HeatPumps2035 version

mheshammenisy committed:
chore: bump dataset version numbers

mheshammenisy committed:
chore: bump version for rerun

amir9667 committed:
refactoring for schyncronizing all

amir9667 committed:
refactoring for schyncronizing all

amir9667 committed:
changing the version

mheshammenisy committed:
refactor(datasets): unify DatasetSources and DatasetTargets definitions across datasets

mheshammenisy committed:
bump(demandregio) version

mheshammenisy committed:
Bump dataset versions for osmtgmod and substation

mheshammenisy committed:
bump(power_plants): increase version to rerun

mheshammenisy committed:
Bump RenewableFeedin dataset version

mheshammenisy committed:
Bump Era5 and RenewableFeedin dataset versions

mheshammenisy committed:
fix Vg250MvGridDistricts bug

mheshammenisy committed:
Bump version: RunPypsaEur

mheshammenisy committed:
Bump version: RunPypsaEur

mheshammenisy committed:
Refactor spatial join implementation

ClaraBuettner committed:
Merge branch 'dev' into features/#1283-fill-dataset-sources-and-targets

ClaraBuettner committed:
Import zensus data from database backup again

In the dataset sources and targets the old way to download the data was still implemented.

ClaraBuettner committed:
Remove unused imports

ClaraBuettner committed:
Iterate over the correct list of files

ClaraBuettner committed:
Update source and targets for new version of MaStR data

ClaraBuettner committed:
Refer to new MaStR data

ClaraBuettner committed:
Drop importing old and new MaStR data since only the new one is used

ClaraBuettner committed:
Do not use hard coded path to MaStR data, add it to dataset sources instead

ClaraBuettner committed:
Apply black and isort

ClaraBuettner committed:
Do not use hard coded MaStR data path

ClaraBuettner committed:
Apply black and isort

ClaraBuettner committed:
Parse path to MaStR data via sources of various datasets

ClaraBuettner committed:
Refer to osm data from 2025 in dataset sources and targets

ClaraBuettner committed:
Apply black and isort

ClaraBuettner committed:
Use next_etrago_id function correctly

ClaraBuettner committed:
Use updated OSM data in osmTGmod

ClaraBuettner committed:
Fix accessing dataset sources

ClaraBuettner committed:
Run sanity_checks in parallel

They don't depend on each other.
When they run in parallel, not all of them are stopped if one is failing.

ClaraBuettner committed:
Store datasets sources and targets in seperate table

ClaraBuettner committed:
Add docstrings again

ClaraBuettner committed:
Apply black

ClaraBuettner committed:
Revert changes

ClaraBuettner committed:
Apply black and isort

ClaraBuettner committed:
Update changelog

sagemaso commented:
Hi @ClaraBuettner and @mheshammenisy, I noticed that the CI for this PR is failing even though it has already been merged. Could you please take a look at the [failing job](https://github.com/openego/eGon-data/actions/runs/23184137430/job/67363401494?pr=1373#step:5:3728)? Thanks!

ClaraBuettner commented:
Hi @sagemaso,
I know that this was merged although the CI tests were partly failing. But the error is not coming from the changes in this branch, it was also on dev and other PRs before.
So @CarlosEpia and I discussed to merge this branch, because the problems are not related and waiting for the fix would have made things more complicated for other further developments.
